crucible-mir/src/Mir/Intrinsics.hs

```diff
@@ -801,6 +804,121 @@ readMirAggregateWithSymOffset bak iteFn off tpr ag@(MirAggregate totalSize m)
     offsetLit = wordLit sym
     iteFn' = liftIteFnMaybe sym tpr iteFn
 
+    -- Given a list of valid offsets in this aggregate, create a predicate that
+    -- the provided offset appears among them.
```

Comment:
When you say "the provided offset", does that refer to the offset at the start of the `NonEmpty` list?